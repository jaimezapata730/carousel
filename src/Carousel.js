import React, { useState } from "react";
import "./Carousel.css";
import image1 from "./image1.jpg";
import image2 from "./image2.jpg";
import image3 from "./image3.jpg";
import Card from "./Card";



 function Carousel( props ) {
  const [cardIdx, setCardIdx] = useState(0);
  const card = props.cardData[cardIdx];
  const total = props.cardData.length;

  //Increments currCardIdx state by 1
  const goForward = () => {
    setCardIdx((prevIdx) => (prevIdx === total -1 ? 0 : prevIdx + 1));
  };

  const goBackward = () => {
    setCardIdx((prevIdx) => (prevIdx === 0 ? total -1 : prevIdx - 1));
  };

  const renderLeftArrow = cardIdx !==0;
  const renderRightArrow = cardIdx !== total -1;

  return (
    <div className="Carousel">
      <h1>{props.title}</h1>
      <div className="Carousel-main">
        { renderLeftArrow && (
          <i
            className="bi bi-arrow-left-circle"
            onClick={goBackward}
            data-testid="left-arrow"
          />
        )}
        <Card
          caption={card.caption}
          src={card.src}
          currNum={cardIdx + 1}
          totalNum={total}
        />
        { renderRightArrow && (
          <i
            className="bi bi-arrow-right-circle"
            onClick={goForward}
            data-testid="right-arrow"
          />
        )}
      </div>
    </div>
  );
}

Carousel.defaultProps = {
  cardData: [
    {
      src: image1,
      caption: 'Image by x'
    },
    {
      src: image2,
      caption: 'Image by y'
    },
    {
      src: image3,
      caption: 'Image by z'
    },
  ],
  title: "Shells from far-away beaches",
};

export default Carousel;
