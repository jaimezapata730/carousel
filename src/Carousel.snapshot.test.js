import React from "react";
import { render } from "@testing-library/react";
import Carousel from "./Carousel";

test('Should match snapshot', function() {
    const { asFragment } = 
    render(<Carousel 
            cardData={[
                { src: 'test1.jpg', caption: 'Caption1' },
                { src: 'test2.jpg', caption: 'Caption2' },
            ]}
            title="Test Carousel"
            />)
    expect (asFragment()).toMatchSnapshot();
});