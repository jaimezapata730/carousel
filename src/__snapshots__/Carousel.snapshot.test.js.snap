// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Test Carousel
    </h1>
    <div
      class="Carousel-main"
    >
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Caption1
        </h4>
        <img
          alt="Caption1"
          class="Card-image"
          src="test1.jpg"
        />
        <small
          class="Card-small"
        >
          Image 1 of 2.
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
        data-testid="right-arrow"
      />
    </div>
  </div>
</DocumentFragment>
`;
